tests/recorder/recorder.go

```diff
@@ -0,0 +1,233 @@
+package main
```

Comment:
Awesome! Got the invite, will take a look now. Thanks! I think it would be really cool to build this library up a bit to support writing test suites around it. For example, I might want to create an in-memory server before running a recorded test to verify that its output matches what the server is sending